src/simulation_strategies/LossyNetworksGeneralizedCliffordsSimulationStrategy.py

```diff
@@ -0,0 +1,26 @@
+__author__ = 'Tomasz Rybotycki'
+
+from typing import List
+
+from numpy import delete, ndarray, vstack, zeros_like
+
+from src.Boson_Sampling_Utilities import prepare_interferometer_matrix_in_expanded_space
+from src.simulation_strategies.GeneralizedCliffordsSimulationStrategy import GeneralizedCliffordsSimulationStrategy
+from src.simulation_strategies.SimulationStrategy import SimulationStrategy
+
+
+class LossyNetworksGeneralizedCliffordsSimulationStrategy(SimulationStrategy):
+    def __init__(self, interferometer_matrix: ndarray) -> None:
+        self._helper_strategy = GeneralizedCliffordsSimulationStrategy(
+            prepare_interferometer_matrix_in_expanded_space(interferometer_matrix))
+
+    def simulate(self, input_state: ndarray) -> List[int]:
+        input_state = input_state.reshape(len(input_state), )  # Divide by two, coz we have 2N x 2N matrix
+        expansion_zeros = zeros_like(input_state)
+        expanded_state = vstack([input_state, expansion_zeros])
+        expanded_state = expanded_state.reshape(2 * len(input_state), )
```

Comment:
This call looks rather weird with `, )`. Is that intended?

Reply:
I know it does, but it is actually  intended. When I put `1` after the comma, then the resultant vector is not in form
`[1, 1, 1, 0, ...]`
but in form
`[[1], [1], [1], [0], ...]`
which causes a lot of problems. I've also seen people using this construction. I imagine there's a way to make this call look better, but given that people use calls like that I decided to let that one slide.